KAN-85-모의면접

kimgorok opened the pull request:
## 😀 제목 : 모의면접 페이지(기업 선택, 자소서 선택, 질문 선택, 모의 면접) 레이아웃 세팅

<br/>

## 🔎 작업 내용

- 어떤 변경 사항이 있었나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

<br/>

## 🎨 이미지 첨부

![image](https://github.com/user-attachments/assets/2658521f-4f1a-44d8-92ed-f760c561d619)

![image](https://github.com/user-attachments/assets/815c039a-a8d2-4e70-8880-2bbcb3743808)

![image](https://github.com/user-attachments/assets/f731c211-f1f9-4fe6-aa04-0a3b33ecb900)

![image](https://github.com/user-attachments/assets/17f3bd0a-04c1-4dca-bd59-fd71680c4e90)

![image](https://github.com/user-attachments/assets/6d4185cf-d533-43db-b9dd-c20438824c23)

![image](https://github.com/user-attachments/assets/e2565516-9e18-4312-b1d2-ca3991c7d149)

![image](https://github.com/user-attachments/assets/be8d8905-1d17-4b5f-a6cd-5b7206d466a1)

![image](https://github.com/user-attachments/assets/708ae625-48f5-4a1a-95b3-57a0e66b52e9)


<br/>

## 🔧 앞으로의 과제

- 다음으로 할 일은 무엇인가요??

아직 모바일 뷰의 UI가 매끄럽지 않습니다. 추후에 헤더가 추가된 후 다시 손보겠습니다.

  <br/>

## ➕ 추가 코멘트
- (10.02)
1. (interview) 폴더에 layout.tsx를 작성했습니다.
2. 기업 선택 페이지를 세팅했습니다.

- (10.03)
1. 질문리스트 페이지를 세팅했습니다.
2. 테스트 json 파일을 작성했습니다.
3. 질문이 생성될 때 애니메이션을 적용했습니다.

- (10.04)
1. 최초질문 페이지를 세팅했습니다.

- (10.05)
1. 최초질문 페이지 UI를 제작중입니다.

- (10.06)
1. 뷰포트의 높이를 측정하는 useContainerHeight.ts 훅을 제작했습니다.
2. 코드 리팩토링을 진행했습니다.

- (10.07)
1. 모의면접 페이지 메시지 전송 함수에 isComposing 검사를 추가해 한글 전송 시 오류를 해결했습니다.
2. ProgressBar.tsx를 생성하고 적용했습니다.
3. 자기소개서 선택 레이아웃을 세팅했습니다.
4. 테스트 json 파일을 작성했습니다.

- (10.09)
1. 채팅 말풍선의 줄바꿈 오류를 수정했습니다.

- (10.10)
1. 코드 리뷰를 반영하여 UI를 수정했습니다.
2. 자소서 선택 페이지와 질문 선택 페이지를 분리하였습니다.

<br/>

## 🤔 Issue Ticket

- [Ticket](https://gwagseoyeon.atlassian.net/browse/KAN-85)

kimgorok committed:
🎉init: [모의면접] layout.tsx 세팅

- 면접 기록 컴포넌트를 layout.tsx에 추가
- 반응형 설정

KAN-85

kimgorok committed:
✨feat: [모의면접] 기업 선택 페이지 레이아웃 세팅

- 돋보기 아이콘 추가 및 적용
- globals.css 커스텀 클래스 추가
- 페이지 세팅 및 반응형 적용

KAN-85

kimgorok committed:
🎉init: [모의면접] 질문 리스트 페이지 세팅

- 질문 리스트 페이지 및 컴포넌트 세팅
- layout UI 수정
- framer-motion 애니메이션 적용
- type 및 테스트 json 생성

KAN-85

kimgorok committed:
🎉init: [모의면접] 최초 질문 페이지 세팅

- 임시 채팅 폼 생성

KAN-85

kimgorok committed:
✨feat: [모의면접] 최초 질문 페이지 진행

- 전반적인 UI 수정
- 화면 크기에 따른 채팅창 UI 변화 적용

KAN-85

kimgorok committed:
♻️refactor: [모의면접] 코드 리팩토링 진행

- ChatContainer 및 MessageForm 컴포넌트 생성
- 뷰 포트의 높이를 측정하는 util 생성
- 전반적인 UI 수정

KAN-85

kimgorok committed:
💄design: [모의면접] css 정리

- globals.css에 클래스를 생성해 UI 정리

KAN-85

kimgorok committed:
✨feat: [모의면접] 하단 스크롤 훅 추가

- useScrollToBottom.ts 추가 및 적용

KAN-85

kimgorok committed:
✨feat: [모의면접] 면접 기록 링크 이동 추가

- 면접 생성 버튼의 링크 이동 추가
- 질문 페이지에 '질문 다시 생성' 버튼 추가

KAN-85

kimgorok committed:
💄design: [모의면접] 전반적인 UI 개선

- 반응형 UI 개선
- globals.css를 포함한 CSS 수정

KAN-85

kimgorok committed:
✨feat: [모의면접] 메시지 전송 함수에 isComposing 검사 추가

- handleKeyDown 함수에 isComposing 검사를 추가해 한글 글자 조합 완료 여부 확인

KAN-85

kimgorok committed:
💄design: [모의면접] 면접 기록 UI 수정

- 면접 기록 크기 조절

KAN-85

kimgorok committed:
✨feat: [모의면접] ProgressBar 생성 및 적용

- ProgressBar.tsx 생성 후 스타일대로 적용

KAN-85

kimgorok committed:
✨feat: [모의면접] 자기소개서 선택 UI 추가

- progressBar 반영
- 테스트 json 작성
- 타입 작성
- 조건부 렌더링 적용

KAN-85

kimgorok committed:
Merge branch 'main' of https://github.com/KEA-5th-Myaong/frontend into KAN-85-모의면접

kimgorok committed:
Merge branch 'KAN-85-모의면접' of https://github.com/KEA-5th-Myaong/frontend into KAN-85-모의면접

kimgorok committed:
💄design: [모의면접] Arrow아이콘 배경색 추가

- ArrowIcons를 사용하는 컴포넌트의 배경색과 테두리색 조정

KAN-85

kimgorok committed:
🐛fix: [모의면접] 채팅 말풍선에 특수문자 줄바꿈 적용

- 채팅 말풍선에 특수문자, 혹은 연속된 알파벳이 입력되었을 때의 줄바꿈 적용

KAN-85

kimgorok committed:
🚚rename: [모의면접] 자기소개서 변수명 변경

- 용어사전에 따라 introduction에서 personalStatement로 변경

KAN-85

yeonilil commented on src/app/(route)/(interview)/_components/ProgressBar.tsx:
```diff
@@ -0,0 +1,15 @@
+      </div>
+    </div>
+  );
+}
```
요기 구현 너무 잘된 것 같아요 👍 
원래 애니메이션 추가했으면 했는데 페이지 넘어가면서 자연스럽게 보여서 좋은 것 같아요

yeonilil replied:
근데 AI 모의면접 진행 후 뒤로가기 버튼을 누르면 질문 선택이 아니라 전전 단계인 자소서 선택으로 가는데 질문 선택도 페이지로 분리하는 건 어떨까요?
글고 이건 그냥 의견인데 ProgressBar 기업 선택, 자소서 선택, 질문 선택 클릭하면 진행한 페이지로 이동하게 해도 사용자가 이용하기 편리할 것 같아요!!

kimgorok replied:
페이지 분리하였습니다!
그런데 ProgressBar를 통한 이동은, 사용자가 기업이나 자소서를 선택하지 않고 다음단계로 이동할 수 있다고 생각하여서 일단은 구현하지 않겠습니다

yeonilil commented on src/app/(route)/(interview)/interview/[id]/[corp]/[questionId]/_components/ChatContainer.tsx:
```diff
@@ -0,0 +1,70 @@
+              {msg.isAI && <p className="font-semibold pb-3">면접관</p>}
+              <div className={`break-words chat-msg-text ${msg.isAI ? 'bg-gray-4' : 'bg-white-2 text-white'}`}>
+                {msg.text}
+              </div>
```
모의면접시 사용자가 보낸 답변 옆에 수정 버튼도 있습니당 추가해주세요~

kimgorok replied:
추가했습니담

yeonilil commented on src/app/(route)/(interview)/interview/[id]/[corp]/[questionId]/_components/MessageForm.tsx:
```diff
@@ -0,0 +1,46 @@
+  return (
+    <form
+      onSubmit={handleSubmit}
+      className="flex items-center py-2 h-[90px] w-full border border-primary-1 rounded-xl mt-auto bg-white-0"
```
MessageForm 부분 처음엔 디자인 시안 크기대로였으면 좋겠습니다! 그리고 내용이 길어지면 input창도 커지도록 하면 더 좋을 것 같아요!
지금 입력창이 조금 크게 있어서요!

kimgorok replied:
반영했습니다!

yeonilil commented on src/app/(route)/(interview)/interview/[id]/[corp]/_components/PersonalStatementBox.tsx:
```diff
@@ -0,0 +1,33 @@
+          <p>수정일자 2024.10.18</p>
+        </div>
+      </div>
+      <div
```
면접 시작 오른쪽 공간 margin이나 혹은 해당 컴포넌트 패딩 조금 더 추가해주시면 좋을 것 같아요! 버튼이 오른쪽에 너무 붙어있어서요!

kimgorok replied:
margin-right : 8px 추가했습니다!

yeonilil commented on src/app/(route)/(interview)/interview/[id]/[corp]/_components/QuestionBox.tsx:
```diff
@@ -0,0 +1,20 @@
+      type="button"
+    >
+      <p className="font-semibold overflow-scroll hide-scrollbar">{question}</p>
+      <div className="flex gap-2 font-semibold text-gray-2 whitespace-nowrap">
```
면접 생성 버튼도 오른쪽 조금만 더 띄워주세요!

kimgorok replied:
margin-right : 8px 추가했습니다!

yeonilil reviewed (approved):
면접 기록 드롭다운도 hover 시 색상 변경하는 대로 구현해주시면 좋을 것 같아요!
전체 색상은 나중에 수정하시는건가요?? 새로 바뀐 색상으로 수정해주세요 😸 

디자인 복잡한 부분도 많았는데 구현하시느라 넘 수고하셨습니당!!

kimgorok committed:
💄design: [모의면접] 코드리뷰 반영 UI 수정

- 면접 기록 hover 배경색 추가
- 채팅 말풍선 크기 수정
- 채팅 입력창 높이 조절
- 면접시작/면접생성 버튼에 margin 추가

KAN-85

kimgorok committed:
🚚rename: [모의면접] (interview)하위 폴더명 변경

- (interview)의 하위 _utils폴더를 _hooks로 변경

KAN-85

kimgorok committed:
♻️refactor: [모의면접] 자소서 선택과 질문선택 페이지 분리

- 기존의 합쳐진 자소서 선택과 질문선택 페이지 분리
- 리스트 애니메이션 모듈로 따로 선언

KAN-85

BaekJiyeon02 commented on src/app/(route)/(interview)/_components/ProgressBar.tsx:
```diff
@@ -0,0 +1,15 @@
+      </div>
+    </div>
+  );
+}
```
이 부분 덕분에 더 퀄리티가 올라가는 것 같아요 구현 잘된 것 같아 좋아요!
혹시 "질문 선택" 순서에서 가운데에서 조금 벗어난 느낌이 있는데 정가운데에 도달할 수 있도록 수정 가능하신가요?😮

kimgorok replied:
넵 반영하겠습니다

BaekJiyeon02 reviewed (commented):
애니메이션을 부드럽게 잘 넣으셔서 예쁜 디자인이 더 업그레이드 되었네요! 너무 좋아요! 고생하셨습니다!
